src/odbc/odbc.c

```diff
@@ -1891,7 +1891,9 @@ SQLCancel(SQLHSTMT hstmt)
 
 	tdsdump_log(TDS_DBG_FUNC, "SQLCancel(%p)\n", hstmt);
 
+        tds_mutex_lock(&stmt->dbc->mtx);
```

Comment:
not sure if SQLCancel is supposed to be also be called in a signal. This would be a deadlock if already locked.

Reply:
Good point. :-/  How would you feel about using `tds_mutex_trylock` and (condtionally on `TDS_HAVE_MUTEX`) directly inspecting the `locked_by` field when unable to (re)take the lock?

Reply:
Never mind, `locked_by` is available only under `ENABLE_EXTRA_CHECKS`.